### ISSUE TITLE
tags in a url?

### ISSUE THREAD

**Issue by RXWatcher** [caronc/apprise-api#83]
:question: **Question**

I'm trying to use apprise api for things like Plex Meta Manager which has it's own json payload.
https://metamanager.wiki/en/latest/config/webhooks.html#run-start-notifications

I'd like to somehow incorporate apprise tags into that so I'm wondering if I could somehow incorporate a tag into the url.
something like this for the apprise api: https://host.com//notify/<key>/?tag=test   ?

Thanks for Apprise. I try to use it everywhere I can :)

-RX

**Comment by caronc** [comment]
Yes, i wrote an [Apprise API](https://github.com/caronc/apprise/wiki/Notify_apprise_api) plugin for Apprise! :slightly_smiling_face: .  It seems a bit counter intuitive, but it supports this exactly.

Try using: `apprise://host.com/?tags=test` and let me know how it goes

**Comment by RXWatcher** [comment]
well, this wouldnt be for apprise (I do run apprise on all of my linux boxes) but to incorporate into places where they want a webhook and I can't control the json payload. 

I use this url as a webhook in DataDog `https://apprise.domain.com/notify/datadog_key`  which does work, it just does a 'all tags'. I can't control the json payload where I could incorporate a tag so I'm basically stuck making a apprise api key for every option. I'd rather do a single key and then use tags.

**Comment by caronc** [comment]
I could add `GET` support for the tags and just consider it overridden if defined in the body/payload?

I don't think that would be a big change

**Comment by RXWatcher** [comment]
yeah, that might work if you're open to adding it.

**Comment by caronc** [comment]
I think i nailed your request; would you be able to test/review it for me and let me know your thoughts?

Is this what you were looking for?

**Comment by RXWatcher** [comment]
yeah, exactly what I'm looking for. It's 11pm here (CEST). I'll give it a
go in the morning and validate that its working.

Thanks again for not only responding to the ticket but also coding up a
solution in short order.

**Comment by RXWatcher** [comment]
I ran a series of tests against it.

1) My use case of DataDog sending the webhook to it with the ?tag= worked. I ran different tag combinations: ?tag=alpha  ?tag=alpha,beta  and it worked in both situations
2) I ran your 2 curl examples and they both worked per your examples. If the ?tag= was added with a previously defined tag then the previously defined tag was not overwritten by the ?tag=. If the ?tag= was added without the previously defined tag it worked as expected.
3) my normal use case of using the same config with apprise and it  worked there as well.

I believe this covers all of my needs. I'm still amazed.

Thanks again for doing this!
